src/modules/dashboard/components/ActionsPage/DetailsWidget/DetailsWidget.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { ReactNode, useState, useEffect } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { useColonyDomainsQuery, AnyToken, Domain } from '~data/index';
+import { Address } from '~types/index';
+import ColorTag from '~core/ColorTag';
+import { getTokenDecimalsWithFallback } from '~utils/tokens';
+import Numeral from '~core/Numeral';
+
+import styles from './DetailsWidget.css';
+
+const displayName = 'dashboard.ActionsPage.DetailsWidget';
+
+const MSG = defineMessages({
+  activeTeam: {
+    id: 'dashboard.ActionsPage.DetailsWidget.activeTeam',
+    defaultMessage: 'Active team',
+  },
+  actionType: {
+    id: 'dashboard.ActionsPage.DetailsWidget.actionType',
+    defaultMessage: 'Action Type',
+  },
+  from: {
+    id: 'dashboard.ActionsPage.DetailsWidget.from',
+    defaultMessage: 'From',
+  },
+  to: {
+    id: 'dashboard.ActionsPage.DetailsWidget.to',
+    defaultMessage: 'To',
+  },
+  value: {
+    id: 'dashboard.ActionsPage.DetailsWidget.value',
+    defaultMessage: 'Value',
+  },
+  paymentActionType: {
+    id: 'dashboard.ActionsPage.DetailsWidget.paymentActionType',
+    defaultMessage: 'Payment',
+  },
+  transferFundsActionType: {
+    id: 'dashboard.ActionsPage.DetailsWidget.transferFundsActionType',
+    defaultMessage: 'Transfer Funds',
+  },
+  recoveryModeActionType: {
+    id: 'dashboard.ActionsPage.DetailsWidget.recoveryModeActionType',
+    defaultMessage: 'Recovery Mode',
+  },
+});
+
+// @TODO we need to add here all possible action types, we need also icons
+export enum ActionTypes {
+  PAYMENT = 'paymentActionType',
+  TRANSFER_FUNDS = 'transferFundsActionType',
+  RECORVERY_MODE = 'recoveryModeActionType',
+}
+
+interface Props {
+  domainId?: number;
+  actionType: ActionTypes;
+  from?: ReactNode;
+  to?: ReactNode;
+  amount?: number;
+  token?: AnyToken;
+  colonyAddress?: Address;
+}
+
+const DetailsWidget = ({
+  domainId,
+  actionType,
+  from,
+  to,
+  amount,
+  token,
+  colonyAddress,
+}: Props) => {
+  const [activeTeam, setActiveTeam] = useState<Domain | undefined>();;
+
+  const { data } = useColonyDomainsQuery({
+    variables: { colonyAddress: colonyAddress || '' },
+  });
+
+  useEffect(() => {
+    if (data) {
+      const domain = data.colony.domains.find(
+        ({ ethDomainId }) => Number(domainId) === ethDomainId,
+      );
+      if (domain) {
+        // Any idea why TS is complaning here?
+        setActiveTeam(domain);
```

Comment:
This is because you are using the wrong domain type for the state. Please use `OneDomain`.

Here's a patch I made for this:
```diff
index 4927e85f1..5d33c1aba 100644
--- a/src/modules/dashboard/components/ActionsPage/DetailsWidget/DetailsWidget.tsx
+++ b/src/modules/dashboard/components/ActionsPage/DetailsWidget/DetailsWidget.tsx
@@ -1,6 +1,6 @@
 import React, { ReactNode, useState, useEffect } from 'react';
 import { defineMessages, FormattedMessage } from 'react-intl';
-import { useColonyDomainsQuery, AnyToken, Domain } from '~data/index';
+import { useColonyDomainsQuery, AnyToken, OneDomain } from '~data/index';
 import { Address } from '~types/index';
 import ColorTag from '~core/ColorTag';
 import { getTokenDecimalsWithFallback } from '~utils/tokens';
@@ -71,7 +71,7 @@ const DetailsWidget = ({
   token,
   colonyAddress,
 }: Props) => {
-  const [activeTeam, setActiveTeam] = useState<Domain | undefined>();;
+  const [activeTeam, setActiveTeam] = useState<OneDomain | undefined>();
 
   const { data } = useColonyDomainsQuery({
     variables: { colonyAddress: colonyAddress || '' },
```

Make sure to apply it to `DetailsWidgetTeam` as well